### DISCUSSION TITLE
3 node swarm with DB persistence - AWS or Digital Ocean?

### DISCUSSION THREAD

**Discussion by craigh1015** [BretFisher/ama#181] [Q&A]
I want to set up a simple 3 node swarm that will initially run node and a persistent store (probably mysql).

Would you recommend Digital Ocean or AWS?

Digital Ocean is far simpler - but it appears multi-node volumes are not directly supported.

AWS is far more complicated and I have not found a recent current video to walk me through the setup process. Docker for AWS is simple - but it produced an 8 node swarm (3+5) rather than the 3 node I am looking for at this stage.

**Comment by BretFisher** [comment]
Options:

1. DO with REX-Ray Swarm Volume manager with [DO driver](https://rexray.readthedocs.io/en/stable/user-guide/schedulers/docker/plug-ins/digitalocean/) for db storage.

2. If you're gonna do AWS and want a smaller footprint, use my [firewall port guide](https://www.bretfisher.com/docker-swarm-firewall-ports/) to make your own EC2 instances. If on AWS, I highly recommend their RDS service for mysql as it's turnkey, saves you a lot of config and testing in your swarm, and keeps your swarm persistent-free hopefully... making it easier to deploy and manage. AWS manages DB's way better then we do anyway :).

**Comment by craigh1015** [comment]
Thanks for the info.

**Comment by BretFisher** [comment]
Keeping open for others to learn
